### ISSUE TITLE
Red arrow for one way collisions of CollisionShape2D pointing in wrong direction in 2D screen

### ISSUE THREAD

**Issue by Daniel-The-Fox** [godotengine/godot#84542]
### Godot version

4.1.2

### System information

Mac OSX 14.1 Sonoma

### Issue description

Hi there, 👋🏻

creating this new issue, as the [issue #12791](https://github.com/godotengine/godot/issues/12791) I found while googling for a solution to my issue with one way collisions is already closed and I was asked to create a new one.

I believe that the big red arrow that appears when enabled the "One Way Collision" checkbox for CollisionShape2D objects in the 2D screen is pointing in the wrong direction and should be inverted. 🤔

When I activate the "One Way Collision" checkbox on a CollisionShape2D the big red arrow points downwards in the 2D screen when the rotation of the CollisionShape2D is 0°. If I rotate the CollisionShape2D by 180°, the big red arrow then shows upwards. So, default seems to be pointing downwards. Ok so far...

The issue is that when I use this CollisionShape2D in a GD script the velocity bounce of my Sprite2D objects is exactly contrary from what I would expect, meaning to the opposite direction in which the big red arrow shows in the 2D View Editor... 😢 🤔

Could you please have a look into this as it's quite confusing for new Godot users like me... 😆 😞

I attached a slim sample project as requested.

### Steps to reproduce

See attached sample project.

### Minimal reproduction project

https://github.com/Daniel-The-Fox/one-way-collision-test

**Comment by Wiltof** [comment]
I fully agree with this, the arrows should show the direction of the one way objects are able to pass through the shape / polygon instead of the opposite. I created a pull request for this: https://github.com/godotengine/godot/pull/84916

There's also space to discuss this during the approval process and so on!

**Comment by kleonc** [comment]
I think the most important part which seems to be ignored here is: it a **red** arrow. Since it's for a one-way collision it would suggest that red color represent the impassable direction, and that seems indeed to be the case:
![w1jAyE17y8](https://github.com/godotengine/godot/assets/9283098/37db6a3f-c0cb-4966-a535-ebe8ef285612)

Looks fine to me.

**Comment by AThousandShips** [comment]
Note also that this has been around for at least 6 years, so changing it is going to throw a lot of people off, and hasn't been a significant issue either seemingly

**Comment by Daniel-The-Fox** [comment]
> I think the most important part which seems to be ignored here is: it a **red** arrow. Since it's for a one-way collision it would suggest that red color represent the impassable direction, and that seems indeed to be the case: ![w1jAyE17y8](https://user-images.githubusercontent.com/9283098/283093923-37db6a3f-c0cb-4966-a535-ebe8ef285612.gif)
> 
> Looks fine to me.

I actually understand this the exact other way around: The red arrow should indicate in which direction the "bounce" force pushes... 🤔😆

Especially the right side in your example is confusing for me, as the box doesn't collide with the still one, although it has a red arrow showing upwards. The logic you mention, would mean that the box on the right should only be able to pass from below to top. In the right case therefore, two arrows or an arrow with two ends would be needed...

**Comment by Ayush-Agarwal123** [comment]
I have a project where a bat hits a ball. The bat is rotated via animation, the collision is expected to be one way collision.
I have the following setup:

This is the hierarchy for final product (Notice this one way collision has red arrow)
![Screenshot 2024-01-11 at 15 47 48](https://github.com/godotengine/godot/assets/76933825/a2243bee-16bd-4323-94e1-68885480d973)
![Screenshot 2024-01-11 at 15 47 56](https://github.com/godotengine/godot/assets/76933825/501ef71a-a167-48b7-a2cd-ce76ddc054ef)

This is the hierarchy for a debug setup in the same project (Notice this one way collision has green arrow)
![Screenshot 2024-01-11 at 15 48 30](https://github.com/godotengine/godot/assets/76933825/437c4120-e0f8-49a0-8c3c-f4dc2e254bc0)
![Screenshot 2024-01-11 at 15 48 37](https://github.com/godotengine/godot/assets/76933825/a222d061-c78a-4e93-be9a-8f0ab6e71d4d)


The issue I am facing is that one way collision is working properly with the debug setup but not with the actual one:
Observations:
In the final setup one way collision is working only when the ball hits the edge (I have also tried rotating the collider)

I am not sure what is causing the issue, I just want to understand is it due to the difference between green and red arrow

My understanding: Green arrow is the direction it lets pass and red arrow is the direction it blocks
